pkg/sessionctx/variable/session.go

```diff
@@ -3650,14 +3650,15 @@ func (s *SessionVars) UseLowResolutionTSO() bool {
 }
 
 // PessimisticLockEligible indicates whether pessimistic lock should not be ignored for the current
-// statement execution. There are cases the `for update` clause should not take effect, like autocommit
-// statements with “pessimistic-auto-commit disabled.
+// statement execution. There are cases the `for update` clause should not take effect, like being
+// executed in an autocommit session.
 func (s *SessionVars) PessimisticLockEligible() bool {
```

Comment:
I'm thinking how we define the relationship between `PessimisticLockEligible` and `PrepareTxnCtx`(when it decides txn mode).
Is there duplicate code/logic/knowledge shared between this function and `PrepareTxnCtx`? Should we consider somehow attain single source of truth by merging them?